web/modules/custom/dpl_related_content/dpl_related_content.module

```diff
@@ -0,0 +1,292 @@
+  // events with the same location.
+  if ((count($event_ids) + count($article_ids) < 4) && ($entity instanceof EventInstance) && !empty($branches)) {
+
+    $event_ids = _dpl_related_content_get_eventinstance_ids($current_uuid, $tags, $categories, $branches);
+  }
+
+  // If the count is still under 4, we'll find the upcoming events, and the
+  // latest articles instead.
+  if (count($event_ids) + count($article_ids) < 4) {
+    $event_ids = _dpl_related_content_get_eventinstance_ids($current_uuid);
+    $article_ids = _dpl_related_content_get_article_ids($current_uuid);
+  }
+
+  // If we still have less than 4, we just won't display anything.
+  if (count($event_ids) + count($article_ids) < 4) {
+    return [];
+  }
+
+  $events = EventInstance::loadMultiple($event_ids);
+  $articles = Node::loadMultiple($article_ids);
+
+  // We want to combine events and articles into a single array, however,
+  // we want to interlace them, so it becomes mixed, rather than just
+  // all the articles and then all the events.
+  // We could also sort the merging based on the actual date values, but this
+  // will be too expensive for what it is worth.
+  $content = [];
+
+  // Remove array keys, so we can do the sorting.
+  $events = array_values($events);
+  $articles = array_values($articles);
+
+  $node_view_builder = \Drupal::entityTypeManager()->getViewBuilder('node');
+  $event_view_builder = \Drupal::entityTypeManager()->getViewBuilder('eventinstance');
+
+  // First, we find the longer array.
+  $length = max(count($events), count($articles));
+
+  for ($i = 0; $i < $length; $i++) {
+    if (isset($events[$i])) {
+      $content[] = $event_view_builder->view($events[$i], 'card');
+    }
+
+    if (isset($articles[$i])) {
+      $content[] = $node_view_builder->view($articles[$i], 'card');
+    }
+  }
+
+  return $content;
+}
+
+/**
+ * Place a dynamic 'related content' list on articles and eventinstances.
+ *
+ * Content displayed is based on the field values of the current page entity.
+ */
+function dpl_related_content_preprocess_page(array &$variables): void {
+  $route_name = \Drupal::routeMatch()->getRouteName();
+
+  // We only want to display the related content slider on eventinstances, OR
+  // on article nodes.
+  if ($route_name === 'entity.node.canonical') {
+    $entity = \Drupal::routeMatch()->getParameter('node');
+
+    if (!($entity instanceof NodeInterface) || $entity->bundle() !== 'article') {
+      return;
+    }
+  }
+  elseif ($route_name === 'entity.eventinstance.canonical') {
+    $entity = \Drupal::routeMatch()->getParameter('eventinstance');
+  }
+  else {
+    return;
+  }
+
+  if (!($entity instanceof FieldableEntityInterface)) {
+    return;
+  }
+
+  try {
+    $variables['related_content'] = _dpl_related_content_get_content($entity);
+  }
+  catch (\Exception $e) {
+    \Drupal::logger('dpl_related_content')->error('Could not render related content slider.');
+  }
+}
```

Comment:
Please include information from the error being thrown.

Otherwise it is impossible to know what has failed.